appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -209,6 +209,13 @@
    */
   protected final Map<String, OptionList> optionLists = Maps.newTreeMap();
 
+  /**
+   * A list of asset filters, which are in fact lists of strings. Initialized with an empty filter.
+   */
+  protected final List<List<String>> filters = new ArrayList<List<String>>() {{
```

Comment:
Rather than create an anonymous inner type here subclassing ArrayList, you could instead break the initialization into a separate `static` block:

```java
protected final List<List<String>> filters;

static {
  filters = new ArrayList<String>();
  filters.add(new ArrayList<String>());
}
```

IMO it's more understandable to most people than the double-curly notation.

Reply:
I wasn't able to add it inside a `static` block b/c the list isn't static. So I moved it to a constructor for the ComponentProcessor instead.